src/Features/CSharp/Portable/CodeCleanup/CSharpCodeCleanupService.cs

```diff
@@ -192,55 +185,42 @@ internal class CSharpCodeCleanupService : ICodeCleanupService
             return document;
         }
 
-        private async Task<Document> ApplyCodeFixesForSpecificDiagnosticIds(
-            Document document, ImmutableArray<string> diagnosticIds, CancellationToken cancellationToken)
+        private async Task<Document> ApplyCodeFixesForSpecificDiagnosticIdsAsync(
+            Document document, ImmutableArray<string> diagnosticIds, IProgressTracker progressTracker, CancellationToken cancellationToken)
         {
             foreach (var diagnosticId in diagnosticIds)
             {
                 using (Logger.LogBlock(FunctionId.CodeCleanup_ApplyCodeFixesAsync, diagnosticId, cancellationToken))
                 {
-                    document = await ApplyCodeFixesForSpecificDiagnosticId(
-                        document, diagnosticId, cancellationToken).ConfigureAwait(false);
+                    document = await _codeFixServiceOpt.ApplyCodeFixesForSpecificDiagnosticIdAsync(
+                        document, diagnosticId, progressTracker, cancellationToken).ConfigureAwait(false);
                 }
             }
 
             return document;
         }
 
-        private async Task<Document> ApplyCodeFixesForSpecificDiagnosticId(
-            Document document, string diagnosticId, CancellationToken cancellationToken)
+        public EnabledDiagnosticOptions GetAllDiagnostics()
         {
-            var tree = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
-            var textSpan = new TextSpan(0, tree.Length);
-
-            var fixCollection = await _codeFixServiceOpt.GetDocumentFixAllForIdInSpan(
-                document, textSpan, diagnosticId, cancellationToken).ConfigureAwait(false);
-            if (fixCollection == null)
-            {
-                return document;
-            }
-
-            var fixAllService = document.Project.Solution.Workspace.Services.GetService<IFixAllGetFixesService>();
-
-            var solution = await fixAllService.GetFixAllChangedSolutionAsync(
-                fixCollection.FixAllState.CreateFixAllContext(new ProgressTracker(), cancellationToken)).ConfigureAwait(false);
-
-            return solution.GetDocument(document.Id);
+            var diagnosticSets = _optionDiagnosticsMappings.SelectAsArray(i => i.diagnosticSet);
+            return new EnabledDiagnosticOptions(diagnosticSets, new OrganizeUsingsSet(true, true));
```

Comment:
1. named parameters for 'true'?  
2.  Now i'm really confused, because there's also not an EnabledDiagnosticOoptions... i don't really undertand why we need all these different ways to do this...

Reply:
1. Now fixed.
2. The underlying limitation is some cleanup operations are triggered by specific diagnostic IDs (and their associated code fixes), while others are triggered by separate configuration. I updated the documentation as well as making the types `sealed` to improve maintainability of this internal API that we aren't 100% happy with.